src/components/App/index.tsx

```diff
@@ -1,187 +1,44 @@
-      case browserPNG.type: return browserPNG.encode(image, encodeData.options);
-      case browserJPEG.type: return browserJPEG.encode(image, encodeData.options);
-      case browserWebP.type: return browserWebP.encode(image, encodeData.options);
-      case browserGIF.type: return browserGIF.encode(image, encodeData.options);
-      case browserTIFF.type: return browserTIFF.encode(image, encodeData.options);
-      case browserJP2.type: return browserJP2.encode(image, encodeData.options);
-      case browserBMP.type: return browserBMP.encode(image, encodeData.options);
-      case browserPDF.type: return browserPDF.encode(image, encodeData.options);
-      default: throw Error(`Unexpected encoder ${JSON.stringify(encodeData)}`);
-    }
-  })();
-
-  const encoder = encoderMap[encodeData.type];
-
-  return new Fileish(
-    [compressedData],
-    sourceFilename.replace(/.[^.]*$/, `.${encoder.extension}`),
-    { type: encoder.mimeType },
-  );
-}
-
-async function processSvg(blob: Blob): Promise<HTMLImageElement> {
-  // Firefox throws if you try to draw an SVG to canvas that doesn't have width/height.
-  // In Chrome it loads, but drawImage behaves weirdly.
-  // This function sets width/height if it isn't already set.
-  const parser = new DOMParser();
-  const text = await blobToText(blob);
-  const document = parser.parseFromString(text, 'image/svg+xml');
-  const svg = document.documentElement;
-
-  if (svg.hasAttribute('width') && svg.hasAttribute('height')) {
-    return blobToImg(blob);
-  }
-
-  const viewBox = svg.getAttribute('viewBox');
-  if (viewBox === null) throw Error('SVG must have width/height or viewBox');
-
-  const viewboxParts = viewBox.split(/\s+/);
-  svg.setAttribute('width', viewboxParts[2]);
-  svg.setAttribute('height', viewboxParts[3]);
-
-  const serializer = new XMLSerializer();
-  const newSource = serializer.serializeToString(document);
-  return blobToImg(new Blob([newSource], { type: 'image/svg+xml' }));
+  file?: File | Fileish;
+  Compress?: typeof Compress;
 }
 
 export default class App extends Component<Props, State> {
-  widthQuery = window.matchMedia('(min-width: 500px)');
-
   state: State = {
-    loading: false,
-    images: [
-      {
-        preprocessorState: defaultPreprocessorState,
-        encoderState: { type: identity.type, options: identity.defaultOptions },
-        loadingCounter: 0,
-        loadedCounter: 0,
-        loading: false,
-      },
-      {
-        preprocessorState: defaultPreprocessorState,
-        encoderState: { type: mozJPEG.type, options: mozJPEG.defaultOptions },
-        loadingCounter: 0,
-        loadedCounter: 0,
-        loading: false,
-      },
-    ],
-    orientation: this.widthQuery.matches ? 'horizontal' : 'vertical',
+    file: undefined,
+    Compress: undefined,
   };
 
   snackbar?: SnackBarElement;
-  readonly encodeCache = new ResultCache();
 
   constructor() {
     super();
+
+    import('../compress').then((module) => {
```

Comment:
Nit: should we handle errors here? It could just use the snackbar to display something about the failure.

Reply:
Yeah, fair point.